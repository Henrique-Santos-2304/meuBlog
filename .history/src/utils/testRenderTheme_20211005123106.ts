import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
