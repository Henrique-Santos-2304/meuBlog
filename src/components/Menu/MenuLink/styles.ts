import styled, { css } from "styled-components";

export const Navigation = styled.div`
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
`;
export const WrapperLink = styled.div`
  ${({ theme }) => css`
    font-size: 1.8rem;
    position: relative;
    letter-spacing: 0.125rem;

    a {
      text-decoration: none;
      padding: 1rem 2rem;
      &:hover {
        :after {
          content: "";
          position: absolute;
          top: 4rem;
          right: 0;

          height: 0.5rem;
          border-radius: 0.6rem;
          background: ${theme.colors.yellow};
          animation: hoverAnimation 0.5s forwards;
        }
      }

      @keyframes hoverAnimation {
        from {
          width: 0;
          right: 50%;
        }
        to {
          width: 100%;
          right: 0;
        }
      }
    }
  `}
`;
export const Link = styled.a``;
