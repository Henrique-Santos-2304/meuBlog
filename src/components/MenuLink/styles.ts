import styled, { css } from "styled-components";

export type MenuIsOpen = {
  isOpen: boolean;
};

export const Wrapper = styled.nav<MenuIsOpen>`
  ${({ theme, isOpen }) => css`
    width: ${theme.grid.containerContent};
    padding: ${theme.spacings.small};
    display: flex;
    justify-content: center;
    align-items: center;
    background: transparent;
    box-shadow: 0 0 30px ${theme.colors.red800}, 0 0 10px ${theme.colors.white};
    backdrop-filter: blur(0.1rem);
    overflow: hidden;
    opacity: ${isOpen ? 1 : 0};
    transition: 1.2s ease-in-out;
    transform: ${isOpen ? "translateX(0)" : "translateX(100%)"};
    @media (max-width: 748px) {
      width: ${theme.grid.container};
      flex-direction: column;
      background: ${theme.colors.white};
      width: 100vw;
      height: 80vh;
      border-radius: 0;

      a {
        color: ${theme.colors.mainBg};
        font-size: 2.5rem;
        margin: ${theme.spacings.small} 0;
      }
    }
  `}
`;
export const MenuLink = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-weight: 700;
    margin: 0 ${theme.spacings.xsmall} ;
    padding: 0 0.3rem;
    font-size: ${theme.font.sizes.xlarge};
    letter-spacing: 0.1rem;
    position: relative;
    transition: filter .3s ;

  @media (max-width: 1020px) {
   font-size: ${theme.font.sizes.medium}
}

    &:hover {

      &::after {
        content: "";
        position: absolute;
        bottom: -0.6rem;
        right: 0;
        height: 0.3rem;
        background: ${theme.colors.red800};
        animation: hoverAnimation .4s forwards  ;
      }
      @keyframes hoverAnimation {
        from {
          width: 0;
          right: 50%;
        }
        to {
          width: 100%;
          right: 0;
        }
      }
    }
  }
  `}
`;
