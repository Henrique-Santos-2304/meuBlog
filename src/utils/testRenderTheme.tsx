import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
