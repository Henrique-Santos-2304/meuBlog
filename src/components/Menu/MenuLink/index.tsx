import * as S from "./styles";
import { VStack } from "@chakra-ui/react";
import Link from "next/link";

type ButtonProps = { onClose?: () => void };

const MenuLinks = ({ onClose }: ButtonProps) => (
  <S.Navigation onClick={onClose}>
    <VStack spacing="6.5rem">
      <S.WrapperLink>
        <Link href="/">Inicio</Link>
      </S.WrapperLink>
      <S.WrapperLink>
        <Link href="/projects">Projetos</Link>
      </S.WrapperLink>
      <S.WrapperLink>
        <Link href="/about">Sobre</Link>
      </S.WrapperLink>
      <S.WrapperLink>
        <Link href="/contact">Contato</Link>
      </S.WrapperLink>
    </VStack>
  </S.Navigation>
);

export default MenuLinks;
