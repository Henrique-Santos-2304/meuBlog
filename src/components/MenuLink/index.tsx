import * as S from "./styles";
export interface CheckProps {
  isOpen: boolean;
}
const MenuLink = ({ isOpen }: CheckProps) => (
  <S.Wrapper aria-hidden={!isOpen} isOpen={isOpen}>
    <S.MenuLink href="#">Inicio</S.MenuLink>
    <S.MenuLink href="#project">Projetos</S.MenuLink>
    <S.MenuLink href="#skills">Habilidades</S.MenuLink>
    <S.MenuLink href="#about">Sobre</S.MenuLink>
  </S.Wrapper>
);

export default MenuLink;
