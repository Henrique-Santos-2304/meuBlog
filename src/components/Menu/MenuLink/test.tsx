import MenuLink from ".";
import { renderTheme } from "utils/testRenderTheme";

describe("<MenuLink />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<MenuLink />);
    expect(container).toMatchSnapshot();
  });
});
