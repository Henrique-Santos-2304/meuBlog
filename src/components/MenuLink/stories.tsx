import { Story, Meta } from "@storybook/react";
import MenuLink from ".";

export default {
  title: "MenuLink",
  component: MenuLink,
} as Meta;

export const Basic: Story = (args) => <MenuLink {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <MenuLink {...args} />;
