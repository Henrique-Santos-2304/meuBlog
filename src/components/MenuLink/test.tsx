import MenuLink from ".";
// import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<MenuLink />", () => {
  it("should Menu it's in the display ", () => {
    const { container } = renderTheme(<MenuLink isOpen />);
    expect(container).toMatchSnapshot();
  });
  // it("should Menu it's visible ", () => {
  //   renderTheme(<MenuLink isOpen />);
  //   const menu = screen.getByTestId("menuLinks");
  //   expect(menu.getAttribute("aria-hidden")).toBe("false");
  // });
  // it("should Menu it's visible ", () => {
  //   renderTheme(<MenuLink isOpen={false} />);
  //   const menu = screen.getByTestId("menuLinks");
  //   expect(menu.getAttribute("aria-hidden")).toBe("false");
  // });
});
