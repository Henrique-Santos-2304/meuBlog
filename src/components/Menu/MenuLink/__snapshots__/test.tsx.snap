// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should render  1`] = `
.c0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 1.8rem;
  position: relative;
  -webkit-letter-spacing: 0.125rem;
  -moz-letter-spacing: 0.125rem;
  -ms-letter-spacing: 0.125rem;
  letter-spacing: 0.125rem;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1rem 2rem;
}

.c1 a:hover:after {
  content: "";
  position: absolute;
  top: 4rem;
  right: 0;
  height: 0.5rem;
  border-radius: 0.6rem;
  background: #E8EB33;
  -webkit-animation: hoverAnimation 0.5s forwards;
  animation: hoverAnimation 0.5s forwards;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="chakra-stack css-1epo9ut"
    >
      <div
        class="c1"
      >
        <a
          href="/"
        >
          Inicio
        </a>
      </div>
      <div
        class="c1"
      >
        <a
          href="/projects"
        >
          Projetos
        </a>
      </div>
      <div
        class="c1"
      >
        <a
          href="/skilss"
        >
          Habilidades
        </a>
      </div>
      <div
        class="c1"
      >
        <a
          href="/contact"
        >
          Contato
        </a>
      </div>
    </div>
  </div>
</div>
`;
